### DISCUSSION TITLE
How to fix `Session.create_table error: object has no attribute '_get_object_id'`?

### DISCUSSION THREAD

**Discussion by eudson** [atoti/atoti#558] [Q&A] [answered]
Hi Everyone,

Am trying to use the template project but, when setting it up for my use case I run into this issue (part of the `stack trace` bellow)

```
  File "/Users/eudson/Library/Caches/pypoetry/virtualenvs/app-fNY1Laf7-py3.10/lib/python3.10/site-packages/py4j/protocol.py", line 298, in get_command_part
    command_part = REFERENCE_TYPE + parameter._get_object_id()
AttributeError: 'DimProductTypeColumn' object has no attribute '_get_object_id'
```

Bellow is a screenshot of the tables am creating the table called `dim_product` (lines 76-85) gets created but the table `dim_product_type` (lines 88-97 ) fails.

<img width="776" alt="image" src="https://user-images.githubusercontent.com/5020414/163024601-991de6ef-fa72-481c-81bc-559b94584e00.png">

**Comment by bloussou** [comment, marked as answer]
Hello,
You are missing `.value` in `create_dim_product_type_table()`:

```diff
  def create_dim_product_type_table(session: tt.Session) -> None:
    session.create_table(
      ...
      types={
        DimProductTypeColumn.ID.value: tt.type.INT,
-       DimProductTypeColumn.PRODUCT_ID: tt.type.INT,
+       DimProductTypeColumn.PRODUCT_ID.value: tt.type.INT,
-       DimProductTypeColumn.TYPE_DESCRIPTION: tt.type.STRING,
+       DimProductTypeColumn.TYPE_DESCRIPTION.value: tt.type.STRING,
      }
    )
```

**Reply by tibdex** [reply]
@eudson, if you run the [`poetry run typecheck`](https://github.com/atoti/project-template/blob/a5b3d12f0721ac8d39ce9acf1e7c73f47ca7c211/pyproject.toml#L34) command you'll be able to catch these kind of issues before runtime.
